examples/http3-server.rs

```diff
@@ -0,0 +1,497 @@
+
+                let q = quiche::accept(&scid, odcid, &mut quiche_config).unwrap();
+
+                let client = HTTP3Client {
+                    quiche_conn: q,
+                    http3_conn: None,
+                };
+
+                clients.insert(scid.to_vec(), (src, client));
+
+                clients.get_mut(&scid[..]).unwrap()
+            } else {
+                clients.get_mut(&hdr.dcid).unwrap()
+            };
+
+            // Process potentially coalesced packets.
+            let read = match client.quiche_conn.recv(buf) {
+                Ok(v) => v,
+
+                Err(quiche::Error::Done) => {
+                    debug!("{} done reading", client.quiche_conn.trace_id());
+                    break;
+                },
+
+                Err(e) => {
+                    error!(
+                        "{} recv failed: {:?}",
+                        client.quiche_conn.trace_id(),
+                        e
+                    );
+                    client
+                        .quiche_conn
+                        .close(false, e.to_wire(), b"fail")
+                        .unwrap();
+                    break 'read;
+                },
+            };
+
+            debug!("{} processed {} bytes", client.quiche_conn.trace_id(), read);
+
+            if client.quiche_conn.is_established() {
+                if client.quiche_conn.application_proto() != b"h3-18" {
+                    // TODO a better error code?
+                    client
+                        .quiche_conn
+                        .close(false, 0x0, b"I don't support your ALPNs")
+                        .unwrap();
+                    break;
+                }
+
+                let h3_config = quiche::h3::Config::new(16, 1024, 0, 0).unwrap();
+
+                if client.http3_conn.is_none() {
+                    debug!(
+                        "{} QUIC handshake completed, now trying HTTP/3",
+                        client.quiche_conn.trace_id()
+                    );
+                    let h3_conn =
+                        quiche::h3::accept(&mut client.quiche_conn, &h3_config)
+                            .unwrap();
+
+                    // TODO some sanity checking that H3 conn is ok before
+                    // adding it to the collection
+                    client.http3_conn = Some(h3_conn);
+                }
+            }
+
+            if let Some(http3_conn) = &mut client.http3_conn {
+                match http3_conn.process(client.quiche_conn.as_mut()) {
+                    Ok((stream_id, quiche::h3::Event::Headers(value))) => {
+                        info!(
+                            "{} Got request {:?} on stream id {}",
+                            client.quiche_conn.trace_id(),
+                            value,
+                            stream_id
+                        );
+
+                        let resp = build_response(root_dir, &value);
+
+                        if resp.0.is_empty() {
```

Comment:
`build_response()` should just return an error if it failed.

Reply:
I agree, this was a hacky to get around figuring out using native Result in the application. Tips welcome.